### DISCUSSION TITLE
Remove the usage of apt-key as it has been deprecated

### DISCUSSION THREAD

**Discussion by naioja** [Azure/CloudShell#214] [Ideas]
We should never install any 3rd party key with apt-key add, because it would cause the system to accept signatures from the third-party keyholder on **all** other repositories configured on the system.

Explained in great detail here : https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key, relevant part below:
"_The problem is not a question of appending a key to one big keyring file etc/apt/trusted.gpg vs manually putting single-key keyring files into the directory /etc/apt/trusted.gpg.d/. These two things are equivalent, and doing either one is a huge security risk._

_The problem is that any key you add to either of the above is completely and unconditionally trusted by apt. This means that when installing any package from any repo (including the official distro repos), apt will happily accept the package being signed by any of those trusted keys (whether the key belongs to the repository the package is coming from or not). This weakens the assurance provided by the package signing mechanism against malicous packages being injected into the official Ubuntu mirrors network._"

This recomandation is part of the Debian wiki for installing 3rd party software: https://wiki.debian.org/DebianRepository/UseThirdParty
_"The key MUST be downloaded over a secure mechanism like HTTPS to a location only writable by root. The key MUST NOT be placed in /etc/apt/trusted.gpg.d or loaded by apt-key add."_

Therefore the example line below 
https://github.com/Azure/CloudShell/blob/2ba756a07322d054942406b49c4eaed99c76c7da/linux/base.Dockerfile#L79
should be modified with the following steps:
1. download the actual key to a diffrent directory then the ones mentioned above and dearmor it:
```curl https://deriv.example.net/debian/deriv-archive-keyring.gpg | gpg --dearmor > /usr/share/keyrings/deriv-archive-keyring.gpg```
3. modify the 3rd party repository definition and mention the key used to sign the package:
```
#/etc/apt/sources.list.d/deriv.list
deb [signed-by=/usr/share/keyrings/deriv-archive-keyring.gpg] https://deriv.example.net/debian/ stable main
```

Although not perfect the Microsoft documentation has an almost accurate example of how to install az-cli using the apt package manager: https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-2-step-by-step-installation-instructions. 
The example is not 100% correct because the key is placed in **/etc/apt/trusted.gpg.d/** meaning it will be trusted for all other repositories whereas it should be placed anywhere else on the file system, as per Debian wiki suggestion **/usr/share/keyrings/** would be preferred.

**Comment by maertendMSFT** [comment]
We are in the process of transitioning to the CBL-Mariner image: https://github.com/microsoft/CBL-Mariner

@darrentu, do we want to hold on making any changes here until that transition is complete?

**Comment by maertendMSFT** [comment]
`apt-key` has now been removed
